### PR TITLE
runReturningOne/List fetches all rows at once

### PR DESCRIPTION
**Pull request by rimmington**
Minimal fix for #474.

### CODE REVIEW

**Comment by tathougies** [comment]
Can you please add some postgres tests using cursors? This looks good to me otherwise.

File: beam-postgres/Database/Beam/Postgres/Connection.hs [hunk 1]
```diff
@@ -201,7 +202,23 @@ withPgDebug dbg conn (Pg action) =
...
+      step (PgRunReturning _ (PgCommandSyntax _ syntax) mkProcess next) =
+        do query <- pgRenderSyntax conn syntax
+           dbg (T.unpack (decodeUtf8 query))
+           _ <- Pg.execute_ conn (Pg.Query query)
+
+           let Pg process = mkProcess (Pg (liftF (PgFetchNext id)))
+           runF process next stepReturningNone
+
+      renderExecReturningList :: (FromBackendRow Postgres x) => _ -> PgSyntax -> (Pg (Maybe x) -> Pg a') -> _ -> _
+      renderExecReturningList errMsg syntax mkProcess next =
```
Comment (tathougies) [inline beam-postgres/Database/Beam/Postgres/Connection.hs:221]: If `errMsg` isn't used, why is it here? I would think that any `*returningList` function can return an empty list without issue.

**Comment by rimmington** [comment]
I added a super basic regression test for `runReturningMany`. I'm struggling to think of a test case that could differentiate between the behaviour of `runReturningMany` and `runReturningList`; I think the only differences should be memory usage and latency?

**Review by 3noch — APPROVED** [review]

**Comment by tathougies** [comment]
No need to differentiate between behaviors. Just want to make sure we are set up to catch any regressions in `runReturningMany` in CI.

**Pull request merged**